website/static/js/filepage/index.js

```diff
@@ -95,9 +95,40 @@ var FileViewPage = {
         self.file = self.context.file;
         self.node = self.context.node;
         self.editorMeta = self.context.editor;
-        //Force canEdit into a bool
-        self.canEdit = m.prop(!!self.context.currentUser.canEdit);
-
+        self.file.checkoutUser = null;
+        self.request_done = false;
+        self.isCheckoutUser = function() {
+            $.ajax({
+                method: 'get',
+                url: window.contextVars.apiV2Prefix + 'files' + self.file.path + '/',
+                beforeSend: $osf.setXHRAuthorization
+            }).done(function(resp) {
+                self.request_done = true;
+                self.file.checkoutUser = resp.data.relationships.checkout.links.related.href ? ((resp.data.relationships.checkout.links.related.href).split('users/')[1]).replace('/', ''): null;
+                if ((self.file.checkoutUser) && (self.file.checkoutUser !== self.context.currentUser.id)) {
+                    m.render(document.getElementById('alertBar'), m('.alert.alert-warning[role="alert"]', m('span', [
+                        m('strong', 'File is checked-out.'),
+                        'This file has been checked-out by a ',
+                        m('a[href="/' + self.file.checkoutUser + '"]', 'collaborator'),
+                        '. It beeds to be checked back in before any changes can be made.'
+                    ])));
+                }
+            }).fail(function(resp) {
+                $osf.growl('Error', 'Unable to get check-out status of this file.');
```

Comment:
I'm not sure how this would fail, besides the mentioned waterbutler issue. Should it be logged here, or assume that whenever it might fail, the error will be logged?